### PR TITLE
Move generating_json_schema to `generate` crate

### PR DESCRIPTION
**Pull request by wkazmierczak**
Move generate_json_schema and generate_docs to `generate` crate.
Substitute commands responsible for generating json schema and docs with one command `cargo run generate_from_type` - it updates both docs and schema. Run with `--check` checks if json schema hasn't changed.

### CODE REVIEW

**Review by wkozyra95 — COMMENTED** [review]
Remove generate_docs and generate_json_schema from https://github.com/membraneframework/live_compositor/blob/master/build_tools/nix/package.nix#L64

File: generate/src/bin/generate_from_types/generate_docs.rs [hunk 1]
```diff
@@ -1,21 +1,18 @@
...
 
-mod definition;
-mod docs_config;
-mod document;
-mod markdown;
-mod schema_parser;
-
-fn main() {
+pub fn generate_docs() {
     tracing_subscriber::fmt().init();
```
Comment (wkozyra95) [inline generate/src/bin/generate_from_types/generate_docs.rs:11]: move that init outside to the main function

File: generate/src/bin/generate_from_types/generate_docs.rs [hunk 2]
```diff
@@ -1,21 +1,18 @@
...
-    let docs_path = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("docs/pages/api/generated");
+    let docs_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .parent()
+        .unwrap()
+        .join("docs/pages/api/generated");
```
Comment (wkozyra95) [inline generate/src/bin/generate_from_types/generate_docs.rs:15]:
```suggestion
        .join("../docs/pages/api/generated");
```

File: generate/src/bin/generate_from_types/generate_json_schema.rs [hunk 3]
```diff
@@ -79,7 +77,10 @@ fn generate_schema(mut current_schema: RootSchema, name: &'static str, action: S
     flatten_definitions_with_one_of(&mut current_schema);
 
     let root_dir: PathBuf = ROOT_DIR.into();
-    let schema_path = root_dir.join(format!("schemas/{}.schema.json", name));
+    let schema_path = root_dir
+        .parent()
+        .unwrap()
+        .join(format!("schemas/{}.schema.json", name));
```
Comment (wkozyra95) [inline generate/src/bin/generate_from_types/generate_json_schema.rs:83]:
```suggestion
        .join(format!("../schemas/{}.schema.json", name));
```

**Review by wkozyra95 — APPROVED** [review]

**Review by WojciechBarczynski — COMMENTED** [review]
LGMT

**Pull request merged**